### PR TITLE
first expedition commit

### PR DESCRIPTION
**Pull request by Kaytila**
under action.tests: Unit tests for all actions, mostly used for learning how everything works.

under utils.swing: Tutorial in a separate JPanel and helper class and a messengerService class which will sooner or later contain all references to showMessage, msg and so on, which will try to find the ui if there is one and use logger.debug otherwise.

under expedition: displayenum - made separate class for it, as there was need to reference it.

some general split() additions.

I always call trim() after split().

Also added a few LayoutPositions for additional UI elements.
Removed some logger elements.

Changed reduce in Equipment: 
If the quantity of an equipment is smaller than the amount to be
removed, just set to 0 instead of going negative. Do not see where this
might make sense.

water is being consumed now.

### CODE REVIEW

File: src/net/slashie/expedition/domain/ItemOffshoreExpeditionContainer.java [hunk 1]
```diff
@@ -118,8 +118,9 @@ public String getTypeDescription() {
 	}
 
 	@Override
-	public int getWaterDays() {
-		return getExpedition().getWaterDays();
+	public int getWaterDays() 
+	{		
+		return getExpedition().getOffShoreWaterDays();
```
Comment (slashman) [inline src/net/slashie/expedition/domain/ItemOffshoreExpeditionContainer.java]: Will this work while the expedition is on sea? when will we use getExpedition().getWaterDays ?

File: .classpath [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
+	<classpathentry kind="lib" path="distro/lib/jcurses-x.jar"/>
+	<classpathentry kind="lib" path="distro/lib/libjcsi.jar"/>
+	<classpathentry kind="lib" path="distro/lib/basicPlayer/jl1.0.jar"/>
+	<classpathentry kind="lib" path="distro/lib/basicPlayer/jorbis-0.0.15.jar"/>
+	<classpathentry kind="lib" path="distro/lib/commons-httpclient-3.0.1.jar"/>
+	<classpathentry kind="lib" path="distro/lib/basicPlayer/commons-logging-api.jar"/>
+	<classpathentry kind="lib" path="distro/lib/basicPlayer/tritonus_share.jar"/>
+	<classpathentry kind="lib" path="distro/lib/basicPlayer/javazoom3.0-openJDK-modfix-src.zip"/>
+	<classpathentry kind="lib" path="distro/lib/basicPlayer/vorbisspi1.0.2.jar"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/4"/>
```
Comment (slashman) [inline .classpath]: As far as I can see, is this the only change on the .classpath ? (changing to v 4), as well as adding log4j ?

File: META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -1,5 +0,0 @@
-Manifest-Version: 1.0
```
Comment (slashman) [inline META-INF/MANIFEST.MF]: Why delete this file? AFAIR it's needed for jar packaging

File: distro/res/denzi/uiAssets.xml [hunk 4]
```diff
@@ -108,4 +108,9 @@
 		<image id = "IMG_MORALE_1" bounds = "368,0,16,16"/>
 		<image id = "IMG_MORALE_0" bounds = "384,0,16,16"/>
 	</tileSet>
+<tileSet file = "img/flowers.png">
+		<image id = "FLW_ACONITUM" bounds = "0,0,96,96"/>
```
Comment (slashman) [inline distro/res/denzi/uiAssets.xml]: What are these flowers used for?

File: distro/res/bigDenzi/appearances.xml [hunk 5]
```diff
@@ -171,6 +171,7 @@
 		<appearance id = "CARAVEL" x = "2" y = "12"/>
 		<!-- Corpses -->
 		<appearance id = "DEAD_NATIVE" x = "1" y = "13"/>
+		<appearance id = "PUEBLO" x = "2" y = "13"/>
```
Comment (slashman) [inline distro/res/bigDenzi/appearances.xml]: Are we using this one for something already?

File: distro/scenarios/theNewWorld/culture.properties [hunk 6]
```diff
@@ -339,7 +339,6 @@
 1673,1354,FARME
 696,350,HUNTE
 2023,1812,FARME
-2272,1736,HUNTE
```
Comment (slashman) [inline distro/scenarios/theNewWorld/culture.properties]: Why were these moved?

File: distro/res/oryx/expedition-oryx.ui [hunk 7]
```diff
@@ -59,13 +59,15 @@ POS_SEADAYS = 11,432
 POS_SUPPLIES_TITLE = 11,456
 POS_SUPPLIES = 20,480
 POS_SUPPLIES_MOD = 20,504
+POS_WATER = 20, 528
 POS_LOCATION_Y = 20
 POS_DATE = 175,478
 POS_TIME = 535,478
 POS_VERSION = 720,575
-
+POS_GOLD = 720, 551
```
Comment (slashman) [inline distro/res/oryx/expedition-oryx.ui]: This doesn't seem to be used?

File: src/net/slashie/expedition/ui/oryx/ExpeditionContinuousGFXUISelector.java [hunk 8]
```diff
@@ -203,34 +234,40 @@ public void activate() {
...
+	{
 		super.shutdown();
 		buttonsPanel.setVisible(false);
-		musicButton.setVisible(false);
-		sfxButton.setVisible(false);
-		saveButton.setVisible(false);
-		quitButton.setVisible(false);
-		logButton.setVisible(false);
+
+		/*
```
Comment (slashman) [inline src/net/slashie/expedition/ui/oryx/ExpeditionContinuousGFXUISelector.java]: We can safely delete the comments in this file

File: .settings/org.eclipse.jdt.core.prefs [hunk 9]
```diff
@@ -0,0 +1,298 @@
+eclipse.preferences.version=1
```
Comment (slashman) [inline .settings/org.eclipse.jdt.core.prefs]: It seems like you commit the .gitignore along with .setting/*, you had to send it on a commit first :)

Can you please delete these three files?

File: .classpath (hunk 2 above)
Comment (Kaytila) [inline .classpath]: Korrekt.

File: META-INF/MANIFEST.MF (hunk 3 above)
Comment (Kaytila) [inline META-INF/MANIFEST.MF]: I had imported a jar file instead of referencing it and thought this is coming from there. So, mistake

File: distro/res/denzi/uiAssets.xml (hunk 4 above)
Comment (Kaytila) [inline distro/res/denzi/uiAssets.xml]: They are referenced and otherwise create a NPE. Not sure yet why they are in there.

File: distro/res/bigDenzi/appearances.xml (hunk 5 above)
Comment (Kaytila) [inline distro/res/bigDenzi/appearances.xml]: As soon as I can add it to the png tileset, yes.

File: distro/scenarios/theNewWorld/culture.properties (hunk 6 above)
Comment (Kaytila) [inline distro/scenarios/theNewWorld/culture.properties]: testing purposes, there was a related NPE

File: distro/res/oryx/expedition-oryx.ui (hunk 7 above)
Comment (Kaytila) [inline distro/res/oryx/expedition-oryx.ui]: It is used, I have included the amount of gold in the main UI.

File: .settings/org.eclipse.jdt.core.prefs (hunk 9 above)
Comment (Kaytila) [inline .settings/org.eclipse.jdt.core.prefs]: I have remember out how to delete them from git without deleting them locally. As I have said, it has been some years ...

File: src/net/slashie/expedition/ui/oryx/ExpeditionContinuousGFXUISelector.java (hunk 8 above)
Comment (Kaytila) [inline src/net/slashie/expedition/ui/oryx/ExpeditionContinuousGFXUISelector.java]: right

Comment (slashman) [inline src/net/slashie/expedition/ui/oryx/ExpeditionContinuousGFXUISelector.java]: This may help http://stackoverflow.com/questions/1143796/remove-a-file-from-a-git-repository-without-deleting-it-from-the-local-filesyste

git rm --cached -r mydirectory

File: .settings/org.eclipse.jdt.core.prefs (hunk 9 above)
Comment (slashman) [inline .settings/org.eclipse.jdt.core.prefs]: This may be helpful http://stackoverflow.com/questions/1143796/remove-a-file-from-a-git-repository-without-deleting-it-from-the-local-filesyste

git rm --cached -r mydirectory

File: distro/res/bigDenzi/appearances.xml (hunk 5 above)
Comment (slashman) [inline distro/res/bigDenzi/appearances.xml]: what for?

File: distro/res/denzi/uiAssets.xml (hunk 4 above)
Comment (slashman) [inline distro/res/denzi/uiAssets.xml]: Ah!!! got it, I think we were working in the "botany" facet before every went kaput :) thanks.

File: distro/scenarios/theNewWorld/culture.properties (hunk 6 above)
Comment (slashman) [inline distro/scenarios/theNewWorld/culture.properties]: Hmmm, that's weird... I don't think we should move them because they were based on real life data

File: src/net/slashie/expedition/locations/SpainCastle.java [hunk 10]
```diff
@@ -18,70 +18,69 @@
...
-			".........#.........................#....................d.W",
-			".........#.....#.....o.............#...#..........#.......#",
-			".........####W##########WWWW#########W#######S########W####",
-			"......,,,XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
-			"......,,,XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
-			"......,,,XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
-			
-		}};
-		
+	public SpainCastle()
```
Comment (slashman) [inline src/net/slashie/expedition/locations/SpainCastle.java]: Please don't introduce more formatting changes to this PR :'(

File: src/net/slashie/expedition/locations/SpainCastle.java [hunk 11]
```diff
@@ -110,164 +109,224 @@ public SpainCastle () {
...
 				ret.addItemOffshore(ItemFactory.createItem("CAPTAIN"), 3);
 				ret.addItemOffshore(ItemFactory.createItem("CARPENTER"), 5);
 				ret.addItemOffshore(ItemFactory.createItem("STEEL_SPEAR"), 82);
 				ret.addItemOffshore(ItemFactory.createItem("STEEL_SWORD"), 3);
-				ret.addItemOffshore(ItemFactory.createItem("BREASTPLATE"), 3);
-				ret.addItemOffshore(ItemFactory.createItem("FRESHWATER"), 1000);
+				ret.addItemOffshore(ItemFactory.createItem("BREASTPLATE"), 3);				
 				ret.addItemOffshore(ItemFactory.createItem("EXPLORER"), 2);
 				ret.addItemOffshore(ItemFactory.createItem("BISCUIT"), 15000);
+				ret.addItemOffshore(ItemFactory.createItem("FRESHWATER"), 30000);
```
Comment (slashman) [inline src/net/slashie/expedition/locations/SpainCastle.java]: Is this a testing value (30.000)

File: src/net/slashie/expedition/ui/console/CharExpeditionDisplay.java [hunk 12]
```diff
@@ -82,12 +79,15 @@ public int showTitleScreen() {
 		File saveDirectory = new File("savegame");
 		File[] saves = saveDirectory.listFiles(new GameFiles.SaveGameFilenameFilter() );
 		boolean loadDisabled = false;
+		if (saves != null)
```
Comment (slashman) [inline src/net/slashie/expedition/ui/console/CharExpeditionDisplay.java]: This is fixed by creating a "savegame" directory on distro, it was missed from git on the transition

File: distro/res/bigDenzi/appearances.xml (hunk 5 above)
Comment (Kaytila) [inline distro/res/bigDenzi/appearances.xml]: different set of tiles for different set of culture.

File: .settings/org.eclipse.jdt.core.prefs (hunk 9 above)
Comment (Kaytila) [inline .settings/org.eclipse.jdt.core.prefs]: Right, found that, taken case of

File: src/net/slashie/expedition/locations/SpainCastle.java (hunk 11 above)
Comment (Kaytila) [inline src/net/slashie/expedition/locations/SpainCastle.java]: Trying to add the same amount of water to the expedition as for food.

File: distro/scenarios/theNewWorld/culture.properties (hunk 6 above)
Comment (Kaytila) [inline distro/scenarios/theNewWorld/culture.properties]: Alright, readded.

File: distro/res/bigDenzi/appearances.xml (hunk 5 above)
Comment (slashman) [inline distro/res/bigDenzi/appearances.xml]: ahhh ok, makes sense.

Comment (Kaytila) [inline distro/res/bigDenzi/appearances.xml]: Just like in the Original: Hunter, Farmer, pueblo, Aztecs, Inca :)

Von meinem iPhone gesendet

File: distro/scenarios/theNewWorld/culture.properties (hunk 6 above)
Comment (Kaytila) [inline distro/scenarios/theNewWorld/culture.properties]: Found out the reason: There is an NPE because of the negative value, will change to positive number.

**Comment by slashman** [comment]
Changes on this PR have been merged into master as part of #2. For further contributions please follow the newly added contribution guidelines :)
